refactor: Add components to labs module

sahlhoff opened the pull request:
<!-- Thank you for your pull request, please provide a brief summary of what this introduces (mandatory). Please point out any code that may be non-obvious to reviewers by using comments. -->

<!-- Make sure that you've linted your files, written and run unit tests, and filled out or updated documentation (README) -->

## Summary

This is migrating `beta_types`, `menu`, and `header` over to `_labs`.

Closes #169 
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

## Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code has been documented and, if applicable, usage described in README.md
- [x] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [x] design approved final implementation
- [x] a11y approved final implementation
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

## Breaking Changes

- `header` is now a `_labs` component. The npm namespace has been updated to: `@workday/canvas-kit-labs-react-header`
- `menu` is now a `_labs` component. The npm namespace has been updated to: `@workday/canvas-kit-labs-react-menu`
- `beta_types.ts` is now a `_labs` component. This component has been moved into `_labs/core` and the npm namespace has been updated to: `@workday/canvas-kit-labs-react-core`
- `space.ts` is now a `_labs` component. This component has been moved into `_labs/core` and the npm namespace has been updated to: `@workday/canvas-kit-labs-react-core`

sahlhoff commented on create-module.sh:
```diff
@@ -228,7 +230,7 @@ packageJson="$cssPath/package.json"
 cat > $packageJson << EOF
 {
-	"name": "@workday/canvas-kit-css-$name",
+	"name": "@workday/canvas-kit-css-$moduleName",
```
These module names are still incorrect

sahlhoff replied:
I've got local changes that will address these issues, lmk if we want it committed on this branch.

sahlhoff replied:
#211 addresses these issues.

NicholasBoll commented:
I'm a bit confused. @anicholls opened #211. Which one should we work on merging?

sahlhoff commented:
@NicholasBoll Alex's PR addresses issues pertaining to `create-module`– this PR is only migrating `beta_types` and `menu` over to `_labs/`. We should merge #211 in first and disregard my changes to `create-module.sh`.

anicholls commented on modules/_canvas-kit-css/index.scss:
```diff
@@ -9,7 +9,6 @@ $wdc-system-icons-path: '../../node_modules/@workday/canvas-system-icons-web/dis
 @import '~@workday/canvas-kit-css-icon/index.scss';
 @import '~@workday/canvas-kit-css-layout/index.scss';
 @import '~@workday/canvas-kit-css-loading-animation/index.scss';
-@import '~@workday/canvas-kit-css-menu/index.scss';
```
@sahlhoff Do you mind moving the css menu back? Sorry - should have been more clear with this one. We want to keep CSS menu available regularly and just move the React menu into Labs. The styling is pretty set in stone, it's just the implementation details of the React menu that we're still figuring out.

sahlhoff replied:
Ah, no worries. Probably should have checked before. 

Question on menus, https://design.workday.com/components/popups/menus, should we update the iframe to point to css if it is the correct design?

sahlhoff commented:
@anicholls the original issue for moving header to labs is blocked by #157 – is that still the case?

lychyi commented:
@sahlhoff can you update this branch? `create-module.sh` went bye-bye in #211

sahlhoff commented:
@lychyi awesome, will do.

lychyi commented on modules/_labs/core/react/package.json:
```diff
@@ -0,0 +1,39 @@
+{
```
Should add these to deps
```
@workday/canvas-colors-web
@workday/canvas-kit-react-core
```

lychyi commented on modules/_labs/header/react/README.md:
```diff
@@ -1,4 +1,8 @@
-# Canvas Kit Header
+# Canvas Kit Labs React Header
```
There are a few other places that reference `@workday/canvas-kit-react-header`

lychyi commented on modules/_canvas-kit-react/index.ts:
```diff
@@ -9,7 +9,6 @@ export * from '@workday/canvas-kit-react-color-picker';
 export * from '@workday/canvas-kit-react-common';
```
Things moved from labs should also be removed from the universal module's `package.json`

sahlhoff commented on modules/core/react/index.ts:
```diff
@@ -33,11 +30,9 @@ export {
-  space,
-  SpaceProps,
   spacingNumbers,
+  stripUnit,
```
@anicholls @lychyi you guys think we should move this into `common/lib/utils/stripUnit.ts`?

lychyi replied:
I think it's fine as is. Otherwise we'll have to add `common` as a dep for `core` right? I think it can be refactored if other packages need it (outside of labs).
